src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,510 @@
+    os.makedirs(run_dir, exist_ok=True)
+
+    # Check if orbit are provided, if Not download
+    if orbit_dir is None:
+        info.log('Orbit directory not assigned. Download orbit ephemerides')
+
+        # Create orbit dir and check internet connection
+        orbit_dir = f'{work_dir}/orbits'
+        os.makedirs(orbit_dir, exist_ok=True)
+        check_internet_connection()
+
+        # List all zip file and extract info
+        zip_file_list = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+
+        for zip_file in zip_file_list:
+            sensor_id, _, start_datetime, \
+            end_datetime, _ = parse_safe_filename(zip_file)
+
+            # Find precise orbits first
+            orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                        end_datetime, 'AUX_POEORB')
+            # If orbit_dict is empty, precise orbits have not been found. Find restituted orbits instead
+            if orbit_dict == None:
+                orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                            end_datetime, 'AUX_RESORB')
+            # Download the orbit file
+            download_orbit(orbit_dir, orbit_dict['orbit_url'])
+
+    # Find reference date and construct dict for geocoding
+    if ref_date is None:
+        ref_file = sorted(glob.glob(f'{slc_dir}/S1*zip'))[0]
+    else:
+        ref_file = glob.glob(f'{slc_dir}/S1*{str(ref_date)}*zip')[0]
+
+    # Generate burst map and prune it if a list of burst ID is provided
+    burst_map = generate_burst_map(ref_file, orbit_dir, x_spac, y_spac, epsg)
+
+    if burst_id is not None:
+        burst_map = prune_dataframe(burst_map, 'burst_id', burst_id)
+
+    # Start to geocode bursts
+    zip_file = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+    for safe in zip_file:
+        i_subswath = [1, 2, 3]
+        orbit_path = get_orbit_file_from_list(safe,
+                                              glob.glob(f'{orbit_dir}/S1*'))
+
+        for subswath in i_subswath:
+            bursts = load_bursts(safe, orbit_path, subswath)
+            for burst in bursts:
+                if burst.burst_id in list(set(burst_map['burst_id'])):
+                    runconfig_path = create_runconfig(burst, safe, orbit_path,
+                                                      dem_file,
+                                                      work_dir, burst_map, flatten,
+                                                      is_split_spectrum,
+                                                      low_band, high_band, pol,
+                                                      x_spac, y_spac)
+                    date_str = burst.sensing_start.strftime("%Y%m%d")
+                    runfile_name = f'{run_dir}/run_{date_str}_{burst.burst_id}.sh'
+                    with open(runfile_name, 'w') as rsh:
+                        path = os.path.dirname(os.path.realpath(__file__))
+                        rsh.write(
```

Comment:
This currently creates many run files which each of them has only one command to run. An alternative would be to have one run file with many commands at each line. The latter allows to run multiple commands in parallel for example using [this](https://github.com/isce-framework/isce2/blob/main/contrib/stack/topsStack/run.py). I wonder if it is preferred to have one run file with one command per line or multiple run files with one command per file?
Maybe @seongsujeong has a preference?

Reply:
@seongsujeong before I make any modification, what is your opinion with respect to the comment above?

Reply:
I've been running them in parallel like 
```
ls run_files  | xargs -P 10 sh -c 'bash run_files/$0 > logfile-$0.log'
```
so for now I've kept them as separate files